File: Searching/SentinelLinearSearch/Recursive/SLS.java
package Searching.SentinelLinearSearch.Recursive;

public class SLS {
    public static int recursiveSearch(int[] arr, int key, int index) {
        if (arr[index] == key)
            return index;
        return recursiveSearch(arr, key, index + 1);
    }

    public static int sentinelLinearSearch(int[] arr, int key) {
        int n = arr.length;
        int last = arr[n - 1];
        arr[n - 1] = key;
        int index = recursiveSearch(arr, key, 0);
        arr[n - 1] = last;
        if (index < n - 1 || arr[n - 1] == key)
            return index;
        return -1;
    }

    public static void main(String[] args) {
        int[] arr = {15, 22, 18, 35, 40};
        int key = 35;
        int index = sentinelLinearSearch(arr, key);
        if (index != -1)
            System.out.println("Element found at index " + index);
        else
            System.out.println("Element not found.");
    }
}

File: Searching/SentinelLinearSearch/Recursive/SLS.py
def recursive_search(arr, key, index):
    if arr[index] == key:
        return index
    return recursive_search(arr, key, index + 1)

def sentinel_linear_search(arr, key):
    n = len(arr)
    last = arr[-1]
    arr[-1] = key
    index = recursive_search(arr, key, 0)
    arr[-1] = last
    if index < n - 1 or arr[-1] == key:
        return index
    return -1

arr = [15, 22, 18, 35, 40]
key = 35
index = sentinel_linear_search(arr, key)

if index != -1:
    print("Element found at index", index)
else:
    print("Element not found.")